megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java

```diff
@@ -664,6 +689,19 @@ public void prepare() {
         graph.dispose();
     }
 
+    /** 
+     * Returns true when an indicator should be shown that some of the stacked
+     * units in the hex of this entity's may still move even while the topmost
+     * shown unit may already have moved. 
+     */
+    private boolean shouldIndicateNotDone() {
+        return (bv.game.getEntitiesVector(entity.getPosition()).indexOf(entity) == 1)
+                && bv.game.getEntitiesVector(entity.getPosition()).stream()
+                .filter(e -> e.getId() != entity.getId())
+                .filter(e -> e.getFacing() == entity.getFacing())
+                .anyMatch(e -> !e.isDone());
```

Comment:
```suggestion
        return bv.game.getEntitiesVector(entity.getPosition()).stream()
                .filter(e -> (e.getId() != entity.getId()) && (e.getFacing() == entity.getFacing()))
                .anyMatch(e -> !e.isDone());
```
A further improvement on the provided code would be a combination of the filters, and a removal of the default return.

Reply:
I think I'll do this instead:

/** 
     * Returns true when an indicator should be shown that a unit with the same facing
     * as this unit is stacked below it and can still move.
     */
    private boolean shouldIndicateNotDone() {
        var hexEntities = bv.game.getEntitiesVector(entity.getPosition());
        return hexEntities.stream()
                .filter(e -> hexEntities.indexOf(entity) < hexEntities.indexOf(e))
                .filter(e -> e.getFacing() == entity.getFacing())
                .anyMatch(e -> !e.isDone());
    }

Actually the super special double arrow should be shown not only for the topmost unit but for any unit that is above another with the same facing that may still move, so I'll compare the indexOf instead which makes the id test unnecessary. On combining filters I really like leaving them apart even if that costs a line of code; maybe Im getting old but I find streams super useful but generally harder to read than good ole ifs due to all the additional .filter.stream.anyMatch()()() clutter. But keeping all logical steps apart in separate lines mitigates that. 
Sure gotta test this. Can't be that easy.

Reply:
You could have the two compares done using an && on different lines.

I've got no proof, but I'd expect it to be slightly better performing

Reply:
Better performance might be but I'm pretty certain it's irrelevant. I do like separating the filters into lines. Just says "no additional complexity here". (Even with the added clutter of .filter()). I also saw this used like this in all examples so I guess there's merit in it. I'll be obstinate here and keep it like that. Apart from that I think it works now and the merge conflict is gone.